File: tracks/python/anagram.py
def is_anagram(word1, word2):
    word1_lowercase = word1.lower()
    word2_lowercase = word2.lower()
    word1_sorted = ''.join(sorted(word1_lowercase))
    word2_sorted = ''.join(sorted(word2_lowercase))
    return word1_sorted == word2_sorted

def find_anagrams(word, candidates):
    word_lowercase = word.lower()
    anagrams = []
    for candidate in candidates:
        if word_lowercase != candidate.lower() and is_anagram(word_lowercase, candidate.lower()):
            anagrams.append(candidate)
    return anagrams

File: tracks/javascript/anagram.js
const isAnagram = (word1, word2) => {
    const word1Lower = word1.toLowerCase();
    const word2Lower = word2.toLowerCase();

    if (word1Lower.length == word2Lower.length && word1Lower != word2Lower) {
        const sortedword1 = word1Lower.split("").sort().join("");
        const sortedword2 = word2Lower.split("").sort().join("");
        return sortedword1 == sortedword2;
    }
}

export const findAnagrams = (word, candidates)  => {
    return candidates.filter(candidate => isAnagram(candidate, word));
};